.github/workflows/goreleaser-test.yml

```diff
@@ -0,0 +1,135 @@
+name: GoReleaser Test
+
+on:
+  workflow_dispatch:
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build binaries with goreleaser
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+
+      - name: Setup Zig
+        uses: goto-bus-stop/setup-zig@v2
+
+      - name: Run GoReleaser check
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          version: latest
+          args: check
+      
+      - name: Run GoReleaser build
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          version: latest
+          args: build --clean --snapshot
+
+      - name: Upload dist folder
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist_folder
+          path: dist
+          retention-days: 1
```

Comment:
Reminds me that some projects build and push container images to a dedicated docker hub account so other users can easily test PRs.  
Not something necessary to do here, but could be interesting in the future.

Reply:
You can always container building on your fork and get a container on your GitHub registry for every branch starting with fb- 😉

Reply:
The idea is it's more for someone else to be able to easily test any PR on their real setup.  
But yeah it would also include that use case you mention.